Create rule S6426: Exclusive tests should not be commited to version control (#985)
SonarSource/rspec: rules/S6426/javascript/rule.adoc

Testing frameworks like Mocha and Jest provide a way to run a single test case for a file by appending `.only()` to the test function. This is convenient when debugging a specific use case. It makes, however, no sense to have it in production or development, and it was likely left by mistake after such a debugging session.

> Therefore, It is strongly recommended not to commit usages of `.only()` to version control.

Reviewer: ```suggestion
Therefore, it is strongly recommended not to commit usages of `.only()` to version control.
```

Reviewer 2: Thanks for the notice, I wonder why it was not caught by grammarly.

> Testing frameworks Mocha and Jest provide a way to run a single test case for a file using the function `.only()`. This is convenient when debugging a specific use case. It makes no sense having it in production or development, but it appears there when cleanup is forgotten after such a debugging session.

Reviewer: ```suggestion
Testing frameworks like Mocha and Jest provide a way to run a single test case for a file by appending `.only()` to the test function. This is convenient when debugging a specific use case. It makes, however, no sense to have it in production or development, and it is likely an oversight after such a debugging session.
```

It is therefore strongly recommended to not commit usages of `.only()` to version control.

== Noncompliant Code Example

[source,javascript]
----
describe("MyClass", function () {
    it.only("should run correctly", function () {
        /*...*/
    });

> });;

Reviewer: ```suggestion
});
```

----

== Compliant Solution

[source,javascript]
----
describe("MyClass", function () {
    it("should run correctly", function () {
        /*...*/
    });

> });;

Reviewer: ```suggestion
});
```
